core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutineAnalyzer.java

```diff
@@ -629,6 +630,8 @@ public Context newScope()
     private static String identifierValue(Identifier name)
     {
         // TODO: this should use getCanonicalValue()
-        return name.getValue();
+        // stop-gap: lowercasing for now to match what is happening during analysis;
+        // otherwise we do not support non-lowercase variables in functions.
+        return name.getValue().toLowerCase(ENGLISH);
```

Comment:
Add a link to https://github.com/trinodb/trino/issues/17 so we can find the places that need to be fixed when we work on that issue.